pip/req/req_install.py

```diff
@@ -805,6 +805,19 @@ def _clean_zip_name(self, name, prefix):
         name = name.replace(os.path.sep, '/')
         return name
 
+    def _get_setup_py_command(self, setup_py_path):
+        cmd = (
+            'import setuptools, tokenize ;'
+            '__file__ = %r ;'
+            '_open = getattr(tokenize, "open", open) ;'
+            'setup_py = _open(__file__).read() ;'
+            'setup_py = setup_py.replace("\\r\\n", "\\n") ;'
+            'codeobj = compile(setup_py, __file__, "exec") ;'
+            'exec(codeobj)'
+        )
```

Comment:
You're right. I didn't think about all the wild cases in which a user's setup.py might break. I could mangle the names, but that would hardly be an improvement over the current:

```python
import setuptools, tokenize;__file__=%r; exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
```

I'll revert to that. Thanks.